indico/modules/events/registration/client/js/form/FormItem.jsx

```diff
@@ -136,6 +154,10 @@ export default function FormItem({
   const inputRequired = !isManagement && showAsRequired;
   const htmlId = `input-${inputProps.fieldId}`;
 
+  const show = useSelector(
+    state => !isItemHidden(state, id) && !isItemHidden(state, showIfFieldId)
```

Comment:
Do you have any opinion on whether this is better than the suggestion I did in [this comment](https://github.com/indico/indico/pull/6678#issuecomment-2801999168)? (since that way the logic is all in one place)

PS: I think the `is_field_shown` property needs to include the same logic as well

Reply:
Oh, because I thought your changes were still applied to the codebase and you were merely describing them. Never mind then, your suggestion is much cleaner. I reverted my changes and applied yours in [e485eab](https://github.com/indico/indico/pull/6678/commits/e485eabf7896a4b37f59b05f25a05bd2f263a9c8).

Reply:
> PS: I think the is_field_shown property needs to include the same logic as well

Done in [a3e992c](https://github.com/indico/indico/pull/6678/commits/a3e992c6ebba90054d4560a54408031d6a765832).